backend/schema/validate.go

```diff
@@ -521,6 +522,52 @@ func validateVerbMetadata(scopes Scopes, module *Module, n *Verb) (merr []error)
 			if _, ok := n.Response.(*Unit); !ok {
 				merr = append(merr, errorf(md, "verb %s: cron job can not have a response type", n.Name))
 			}
+		case *MetadataRetry:
+			// Only allow retries on FSM transitions for now
+			fsms := islices.Filter(module.Decls, func(d Decl) bool {
+				fsm, ok := d.(*FSM)
+				if !ok {
+					return false
+				}
+				starts := islices.Filter(fsm.Start, func(ref *Ref) bool {
+					return ref.Name == n.Name
+				})
+				if len(starts) > 0 {
+					return true
+				}
+				transitions := islices.Filter(fsm.Transitions, func(t *FSMTransition) bool {
+					return t.To.Name == n.Name
+				})
+				return len(transitions) > 0
+			})
+			if len(fsms) == 0 {
+				merr = append(merr, errorf(md, "verb %s: retries can only be added to FSM transitions", n.Name))
+				return
+			}
+
+			// Validate count
+			if md.Count != nil && *md.Count <= 0 {
+				merr = append(merr, errorf(md, "verb %s: retry count must be atleast 1", n.Name))
+			}
+
+			// Validate parsing of durations
+			minDuration, err := md.MinBackoffDuration()
+			if err != nil {
+				merr = append(merr, errorf(md, "verb %s: %v", n.Name, err))
+				return
+			}
+			if minDuration == 0 {
+				merr = append(merr, errorf(md, "verb %s: retry duration must be atleast 1s", n.Name))
+				return
+			}
+			maxDuration, err := md.MaxBackoffDuration()
+			if err != nil {
+				merr = append(merr, errorf(md, "verb %s: %v", n.Name, err))
+				return
+			}
+			if maxDuration, ok := maxDuration.Get(); ok && maxDuration < minDuration {
```

Comment:
Should there be a check against `MaxBackoffDuration` (ie. 24h)?

Reply:
Both MinBackoff and MaxBackoff are checked for min/max during `parseRetryDuration`